### PR TITLE
Epidemiological random tips

### PR DESCRIPTION
**Pull request by Faustrox**
**Description:** Doing PDFView for tips on Epidemiological.

PD: We need to comment some code from another file because of images not found, @RutsSantos is working on that.

**Linked issues:** [Ticket](https://trello.com/c/PlS3Ud4B/102-hacer-que-los-consejos-de-salud-mental-se-muestren-aleatoriospdf)

**Screenshots:**
Screenshot_2020-06-03-19-18-10-239_org pathcheck covidsafepaths

**How to test:** On homescreen settings you can see an Epidemiological option... There you can see a tab navigation, go to the second one (to the right one) and that is the PDF... Is programmed to change between days.

### CODE REVIEW

File: app/Entry.js [hunk 1]
```diff
@@ -186,6 +187,8 @@ class Entry extends Component {
             options={{ headerShown: false }}
           />
           <Stack.Screen
+            name='EpidemiologicResponse'
```
Comment (iocedano) [inline app/Entry.js]: Has two `name` and `component`!!

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalStatus.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            {/* <Image
+              resizeMode='contain'
+              style={{
+                height: 200,
+                width: 310,
+                alignSelf: 'center',
+                marginTop: 10,
+              }}
+              source={require('../../../assets/images/waitingResults.jpg')}
+            /> */}
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalStatus.js]: Could we remove this?

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalStatus.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            {/* <Image
+              resizeMode='contain'
+              style={{
+                height: 230,
+                width: 310,
+                alignSelf: 'center',
+                marginTop: 10,
+              }}
+              source={require('../../../assets/images/covidFree.jpg')}
+            /> */}
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalStatus.js]: Same here

File: app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    // Tip con contador
+
+    const newTip = oldTip + 1;
+
+    return newTip >= maxTips ? 1 : newTip;
+  };
+
+  async componentDidMount() {
+    const { date: storageDate, tip: storageTip } = await this.getData();
+    const isNewDate = this.isNewDate(storageDate);
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js]: `isNewDate` confuse by the name is a date or bool

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      return storageData
+        ? JSON.parse(storageData)
+        : { date: undefined, tip: 0 };
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  saveData = async (newDate, newTip) => {
+    try {
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]: Why do you need the try-catch?

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      await SetStoreData('epidemiologicalTips', {
+        date: newDate,
+        tip: newTip,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  };
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]:
```suggestion
  saveData = async (date, tip) => {
    try {
      await SetStoreData('epidemiologicalTips', {
        date,
        tip,
      });
    } catch (err) {
      console.error(err);
    }
  };
```

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    try {
+      const storageData = await GetStoreData('epidemiologicalTips');
+
+      return storageData
+        ? JSON.parse(storageData)
+        : { date: undefined, tip: 0 };
+    } catch (err) {
+      console.error(err);
+    }
+  };
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]:
```suggestion
  getData = async () => {
    const storageData = await GetStoreData('epidemiologicalTips');

    return storageData || { date: Date.now(), tip: 1 };
  };
```

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+      await SetStoreData('epidemiologicalTips', {
+        date: newDate,
+        tip: newTip,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  isNewDate = dateStorage => {
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]: Do really understand the purpose of this function,
Could you help me?

File: app/views/DR/EpidemiologicalResponseScreen/index.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <NavigationBarWrapper
+      title={t('label.epidemiologic_report_title')}
+      onBackPress={() => navigation.goBack()}>
+      <View style={{ flex: 1, backgroundColor: Colors.WHITE }}>
+        <TopBar.Navigator
+          tabBarOptions={{
+            activeTintColor: '#0059ff',
+            inactiveTintColor: '#000',
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/index.js]: Could we use the color for the palette color file?

File: app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js:1]: Duplicate component

**Review by iocedano — CHANGES_REQUESTED** [review]

File: app/Entry.js (hunk 1 above)
Comment (Faustrox) [reply app/Entry.js]: What do you mean?

File: app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js (hunk 10 above)
Comment (Faustrox) [reply app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js:1]: Where?

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js (hunk 8 above)
Comment (Faustrox) [reply app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]: Is for make a condition with today date and the date storage in AsycnStorage

File: app/views/DR/EpidemiologicalResponseScreen/index.js (hunk 9 above)
Comment (Faustrox) [reply app/views/DR/EpidemiologicalResponseScreen/index.js]: Speak to @RutsSantos about that... Those changes come from Rut's branch

File: app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js [hunk 11]
```diff
@@ -1,9 +1,99 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      date: '',
+      tip: '',
+    };
+  }
+
+  getData = async () => {
+    const storageData = JSON.parse(await GetStoreData('epidemiologicalTips'));
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js]: If you send `false` in the second parameter, we don't need to parse the JSON
https://github.com/intellisysdcorp/covid-safe-paths/blob/develop/app/helpers/General.js#L10-L23

```suggestion
    const storageData = await GetStoreData('epidemiologicalTips', false);
```

File: app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js [hunk 12]
```diff
@@ -1,9 +1,99 @@
...
+
+  saveData = async (date, tip) => {
+    return await SetStoreData('epidemiologicalTips', {
+      date,
+      tip,
+    });
+  };
+
+  isTipGoingToChange = dateStorage => {
+    // Aqui va a haber una condicion para saber si es el mismo dia o no
```
Comment (iocedano) [inline app/views/DR/EpidemiologicalResponseScreen/mentalHealthAdvices.js:31]: Could you add the logic for that?

**Review by iocedano — APPROVED** [review]

**Comment by iocedano** [comment]
They are things we may need to improve later
SGTM

File: app/Entry.js (hunk 1 above)
Comment (iocedano) [reply app/Entry.js]: The component has two attributes `name` and two `component`

File: app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js (hunk 8 above)
Comment (iocedano) [reply app/views/DR/EpidemiologicalResponseScreen/epidemiologicalTips.js]: Ok

File: app/Entry.js (hunk 1 above)
Comment (Faustrox) [reply app/Entry.js]: Is necessary... The name is not going to show up though,

**Review by inteljudajocu — APPROVED** [review]
Improvement has to be made but it LGTM

**Pull request merged**